### ISSUE TITLE
Stage 1 pretraining on YouTubeASL

### ISSUE THREAD

**Issue by pcc03** [ZechengLi19/Uni-Sign#26]
Dear author,

I am writing to ask for your help with the YouTubeASL pertaining phase. I selected about 9000 samples (segmented sentences) for stage1 pertaining. And following the same data split scheme as CSL_News, I split 99% for train and 1% for dev/test.

The problem is that the loss is gradually decreasing, but the test_bleu scores remain low. Also, the test_bleu score keeps decreasing after a certain point as the epoch increases. The specific result for the first several epochs is pasted below.

 I suspect it might be caused by a small training set, with about 9000 samples. However, due to storage limitations, I cannot use the full YouTubeASL dataset. So, what is the minimum training scale required to achieve a similar performance to yours? Could you confirm it or give me some hints about the reason?

Hope you have a good day!

> {"train_lr": 0.0002994021600690321, "train_loss": 11.171271929824561, "test_loss": 6.932291666666667, "test_bleu1": 4.483608688695096, "test_bleu2": 0.7877730574157377, "test_bleu3": 0.3089830300819053, "test_bleu4": 0.16524723608377342, "test_rouge": 3.647281886115711, "epoch": 0, "n_parameters": 587.747368}
> {"train_lr": 0.00029575701895414924, "train_loss": 6.908936403508772, "test_loss": 6.229166666666667, "test_bleu1": 3.265114051961132, "test_bleu2": 0.6348387059426968, "test_bleu3": 0.2587023923966386, "test_bleu4": 0.14045705822846571, "test_rouge": 3.4287668564941853, "epoch": 1, "n_parameters": 587.747368}
> {"train_lr": 0.00028851901975062956, "train_loss": 6.458059210526316, "test_loss": 6.104166666666667, "test_bleu1": 4.697919503850828, "test_bleu2": 0.9386282706567834, "test_bleu3": 0.30971496319489766, "test_bleu4": 0.15148718116414753, "test_rouge": 6.108544154660626, "epoch": 2, "n_parameters": 587.747368}
> {"train_lr": 0.0002778624367725383, "train_loss": 6.249725877192983, "test_loss": 6.041666666666667, "test_bleu1": 7.0007120561033975, "test_bleu2": 1.3498719685604101, "test_bleu3": 0.47798631947433945, "test_bleu4": 0.2442712893068329, "test_rouge": 5.498521643238048, "epoch": 3, "n_parameters": 587.747368}
> {"train_lr": 0.00026405628058170146, "train_loss": 6.124835526315789, "test_loss": 5.989583333333333, "test_bleu1": 7.32810575544929, "test_bleu2": 1.1480972666255604, "test_bleu3": 0.43981568717429165, "test_bleu4": 0.2337859917980113, "test_rouge": 5.202944450235364, "epoch": 4, "n_parameters": 587.747368}
> {"train_lr": 0.00024743678560717466, "train_loss": 6.032072368421052, "test_loss": 5.958333333333333, "test_bleu1": 6.433150483589454, "test_bleu2": 1.2534131797788977, "test_bleu3": 0.4350758795809152, "test_bleu4": 0.21971905282686008, "test_rouge": 5.482180798475602, "epoch": 5, "n_parameters": 587.747368}
> {"train_lr": 0.00022841269720050978, "train_loss": 5.947642543859649, "test_loss": 5.947916666666667, "test_bleu1": 6.026495946310039, "test_bleu2": 1.188463056134231, "test_bleu3": 0.4592447151967451, "test_bleu4": 0.2449597256387031, "test_rouge": 5.7905658822029356, "epoch": 6, "n_parameters": 587.747368}
> {"train_lr": 0.00020745909740887663, "train_loss": 5.887116228070176, "test_loss": 5.994791666666667, "test_bleu1": 5.035357127326339, "test_bleu2": 0.8332913369521655, "test_bleu3": 0.32465336222351604, "test_bleu4": 0.1731450802693395, "test_rouge": 5.631699014395059, "epoch": 7, "n_parameters": 587.747368}
> {"train_lr": 0.000185084502605407, "train_loss": 5.84671052631579, "test_loss": 5.942708333333333, "test_bleu1": 4.165735151650644, "test_bleu2": 0.6453349858609059, "test_bleu3": 0.2470137691490845, "test_bleu4": 0.13002657154287767, "test_rouge": 5.505678440975199, "epoch": 8, "n_parameters": 587.747368}
>

**Comment by ZechengLi19** [comment]
@pcc03 Hi

A more direct explanation is that the training set is likely too small. With only ~9k samples, it is difficult for the model to converge under a standard language modeling objective. In contrast, the YouTube-ASL dataset used in our earlier experiments contains over 500k segments, suggesting that substantially more data is required to obtain a well-performing model.

As I am approaching graduation and there has been considerable interest in the ASL weights, I am re-implementing this part of the pipeline for public release. The original pose-only weights for How2Sign and OpenASL are expected to be released in the near term, no later than January 1, 2026. 

Due to a server migration, the original ASL Stage-1 weights and part of the YouTube-ASL data were lost, and I am therefore attempting to reproduce them under limited computational resources using only a subset of the data. Preliminary results show a BLEU-4 score of approximately 13 on How2Sign; OpenASL has not yet been fully evaluated and is expected to underperform the results reported in the original paper. If you are interested, I can share the recently reproduced ASL Stage-1 pose-only weights.

**Comment by pcc03** [comment]
I see. Thank you! My email is peng326@purdue.edu.

Hope you have a good day.

**Comment by immortal12389** [comment]
> 我明白了，谢谢！我的邮箱是[peng326@purdue.edu](mailto:peng326@purdue.edu)。
> 
> 祝你今天过得愉快。

Hello, I was wondering if you have successfully reproduced the first and second stages of pre-training for YouTubeASL. I am also trying to reproduce it, but my lab equipment is insufficient to support it. If you have completed the reproduction, would you please share the pre-trained weights with me? Thank you very much.
